apollo-runtime/src/main/java/com/apollographql/apollo/internal/ApolloCallTracker.java

```diff
@@ -0,0 +1,174 @@
+package com.apollographql.apollo.internal;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * ApolloCallTracker is responsible for keeping track of running {@link ApolloCall} & {@link ApolloPrefetch} objects.
+ */
+public class ApolloCallTracker {
+
+  private Runnable idleCallback;
+
+  private final Deque<ApolloCall> runningSyncCalls = new ArrayDeque<>();
+  private final Deque<RealApolloCall.AsyncCall> runningAsyncCalls = new ArrayDeque<>();
+  private final Deque<ApolloPrefetch> runningSyncPrefetches = new ArrayDeque<>();
+  private final Deque<RealApolloPrefetch.AsyncCall> runningAsyncPrefetches = new ArrayDeque<>();
```

Comment:
I guess so. Don't really have a strong opinion here but I felt that even though sync and async refer to the same object type, it's better to have different deques for tracking async and sync calls, just to keep concerns separate. We don't have a use case right now though wherein we only need to track all the async operations or only the sync operations. 

If you feel otherwise I'll just go ahead and make the change though.